### ISSUE TITLE
No options to change domain name and target dev for interface

### ISSUE THREAD

**Issue by deepraut89** [vagrant-libvirt/vagrant-libvirt#799]
### Steps to reproduce
I am unable to change the domain name and there seems to be no option to update "target dev" for interface.

```Vagrant.configure("2") do |config|
  config.vm.define :vpc21 do |vpc21|
    vpc21.vm.box = "debian87-64bit"
    config.ssh.insert_key = false
    vpc21.vm.synced_folder ".", "/vagrant", disabled: true
    vpc21.vm.provider :libvirtd do |domain|
       domain.name = "vpc21"
       domain.storage :file, :size => '5G', :type => 'qcow2'
    end
    vpc21.vm.network :public_network,
        :dev => "vpc21_eth1_br",
        :target => "vpc21_eth1",
        :mode => "bridge",
        :type => "bridge",
        auto_config:false
```
Output:
```
# virsh list 
Id    Name                           State
----------------------------------------------------
 325   getting-started_vpc21          running
 337   testing-setup_vpc21            running
# brctl show
vpc21_eth1_br           8000.8e44f671200a       no              21e1
                                                        vnet3
                                                        vnet9
```
### Expected behaviour
Domain name should be configurable.
Target dev for interface should be configurable

### System configuration
**OS/Distro version:**: CentOS-7

**Libvirt version:**2.0.0

**Output of `vagrant version; vagrant plugin list`:**Installed Version: 1.9.7

**Comment by infernix** [comment]
I am not sure what you mean by changing the domain name, are you referring to `getting-started_xxx` or `xxx_vpc21`? the prefix is based on the directory, the suffix is based on whatever you set as `config.vm.define :XXX do`

Secondly, "target dev for interface should be configurable" - there is no such thing as a `target` setting. Here's whats supported:

```
            if iface_configuration[:iface_type] == :public_network
              @device = iface_configuration.fetch(:dev, 'eth0')
              @mode = iface_configuration.fetch(:mode, 'bridge')
              @type = iface_configuration.fetch(:type, 'direct')
              @model_type = iface_configuration.fetch(:model_type, @nic_model_type)
              @driver_name = iface_configuration.fetch(:driver_name, false)
              @driver_queues = iface_configuration.fetch(:driver_queues, false)
              @portgroup = iface_configuration.fetch(:portgroup, nil)
              @network_name = iface_configuration.fetch(:network_name, @network_name)
              template_name = 'public_interface'
              @logger.info("Setting up public interface using device #{@device} in mode #{@mode}")
              @ovs = iface_configuration.fetch(:ovs, false)
              @trust_guest_rx_filters = iface_configuration.fetch(:trust_guest_rx_filters, false)
```

Not quite sure what you want to achieve here anyway?

**Comment by deepraut89** [comment]
Thanks for your reply.
What i mean is I want to override the prefix(current directory name) addition in the name of machine, i.e. it should be "vpc21" only what i defined in Vagrantfile. Though I appreciate this feature, but need to override this based on my current setup.
I am using virtual machines to create some topology, there my vpc21.xml file for interface should show:
```
<interface type='bridge'>
      <mac address='52:54:00:96:3a:b4'/>
      <source bridge='vpc21_eth2_br'/>
      <target dev='vpc21_eth2'/>
      <model type='rtl8139'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x05' function='0x0'/>
    </interface>
```

Same thing I want to achieve using Vagrantfile but I am not getting options to change "<target dev" and "<model type='******'/>" to "rtl8139".

**Comment by infernix** [comment]
`model` type can be changed with `model_type` parameter, but we don't currently have support for a `target` override.

As for the prefix name, looks like we have an undocument `libvirt.default_prefix` setting but it seems broken in its current form as it doesn't get parsed.

**Comment by electrofelix** [comment]
I can't pinpoint quite where the default_prefix behaviour was fixed, however with #885 where the leading `_` was removed it became possible to just use the machine name as the domain name. #908 added it to the docs and #1692 adds the ability to set the target device for public nic interfaces as well, with a configuration validation to reject any of the reserved names upfront.
